lib/service/modules/push.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const request = require('request');
+const req = request.defaults({});
+
+const MISSING_USER_PARAM_ERROR = 'You must specify the \'users\' parameter to send a push notification';
+const MISSING_MESSAGE_PARAM_ERROR = 'You must specify the \'message\' parameter to send a push notification';
+
+module.exports = (proxyURL, taskMetadata, proxyTaskEmitter) => {
+  const _sendMessage = (userEntities, message, callback) => {
+    let proxyShouldWaitForConfirmation = 'true';
+
+    if (typeof callback === 'undefined' || callback === null) {
+      proxyTaskEmitter.emit('proxyTaskStarted');
+      proxyShouldWaitForConfirmation = 'false';
+    }
+    if (typeof userEntities === 'undefined' || userEntities === null) throw new TypeError(MISSING_USER_PARAM_ERROR);
+    if (typeof message === 'undefined' || message === null) throw new TypeError(MISSING_MESSAGE_PARAM_ERROR);
+    if (userEntities.constructor !== Array && typeof userEntities === 'object') userEntities = [userEntities];
```

Comment:
Unlike the other occurrence, changing this to `!util.userEntities.isArray...` broke the code